### PR TITLE
performance: limit was chosen too big for sorted select queries

### PR DESCRIPTION
**Pull request by mfelsche**
it was (limit + (2*offset))

### CODE REVIEW

**Comment by mfelsche** [comment]
example benchmark runs:

```
Running: SELECT "pageURL" FROM rankings_cj ORDER BY "pageURL" LIMIT 1 OFFSET 10000
Runtime: best: 0.271 previous: 0.566 current: 0.324
```
```
Running: SELECT "sourceIP" FROM uservisits_cj ORDER BY "sourceIP" LIMIT 1 OFFSET 50000
Runtime: best: 0.881 previous: 1.539 current: 0.913
```

```
Running: SELECT "sourceIP" FROM uservisits_cj ORDER BY "sourceIP" LIMIT 1 OFFSET 10000
Runtime: best: 0.293 previous: 0.605 current: 0.327
```

File: sql/src/main/java/io/crate/planner/consumer/QueryAndFetchConsumer.java [hunk 1]
```diff
@@ -193,7 +194,7 @@ private PlannedAnalyzedRelation normalSelect(QueriedTableRelation table,
                         toCollect,
                         collectPhaseProjections
                 );
-                collectPhase.limit(projectionBuilderContext.limit);
+                collectPhase.limit(MoreObjects.firstNonNull(projectionBuilderContext.limit, Constants.DEFAULT_SELECT_LIMIT) + projectionBuilderContext.offset);
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/planner/consumer/QueryAndFetchConsumer.java]: Shouldn't we keep it at null instead of setting it to the default limit?

Comment (mfussenegger) [inline sql/src/main/java/io/crate/planner/consumer/QueryAndFetchConsumer.java]: Ah, we did that before too.. so guess it's okay

**Comment by mfussenegger** [comment]
ack

**Pull request merged**